octoprint_mrbeam/analytics/usage_handler.py

```diff
@@ -356,36 +358,59 @@ def _load_usage_data(self):
                 "Trying to recover from _backup_file file: %s", self._backup_file
             )
             recovery_try = True
-            if os.path.isfile(self._backup_file):
+            try:
+                with open(self._backup_file, "r") as stream:
+                    data = yaml.safe_load(stream)
+                if self._validate_data(data):
+                    data["restored"] = (
+                        data["restored"] + 1 if "restored" in data else 1
+                    )
+                    self._usage_data = data
+                    success = True
+                    self._write_usage_data()
+                    self._logger.info("Recovered from _backup_file file. Yayy!")
+            except yaml.constructor.ConstructorError:
                 try:
-                    data = None
-                    with open(self._backup_file, "r") as stream:
-                        data = yaml.safe_load(stream)
-                    if self._validate_data(data):
-                        data["restored"] = (
-                            data["restored"] + 1 if "restored" in data else 1
-                        )
-                        self._usage_data = data
-                        success = True
-                        self._write_usage_data()
-                        self._logger.info("Recovered from _backup_file file. Yayy!")
-                except:
-                    self._logger.error("Can't read _backup_file file.")
+                    success = self._repair_backup_usage_data()
+                except Exception:
+                    self._logger.error("Repair of the _backup_file failed.")
+            except OSError:
+                self._logger.error("There is no _backup_file file.")
+            except yaml.YAMLError:
+                self._logger.error("There was a YAMLError with the _backup_file file.")
+            except:
+                self._logger.error("Can't read _backup_file file.")
 
         if not success:
             self._logger.warn("Resetting usage data. (marking as incomplete)")
             self._usage_data = self._get_usage_data_template()
             if recovery_try:
                 self._write_usage_data()
 
+    def _repair_backup_usage_data(self):
+        success = False
+        with open(self._backup_file, "r") as stream:
+            data = yaml.load(stream)
+        if self._validate_data(data):
+            if isinstance(data["version"], unicode):
+                data["version"] = unicodedata.normalize('NFKD', data["version"]).encode('ascii', 'ignore')
+            data["restored"] = (
+                data["restored"] + 1 if "restored" in data else 1
```

Comment:
what is that line supposed to do ?

Reply:
the restored? It will count how often this file was restored from the backup file